[Santosh Muthukrishnan] iP

Santosh3007 opened the pull request:
# Duke

> "Your mind is for having ideas, not holding them." - David Allen [(source)](https://dansilvestre.com/productivity-quotes)
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [ ] Managing tasks
- [ ] Managing deadlines (Coming soon)
- [ ] Reminders (Coming soon)
***
If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Santosh3007 committed:
feat: read user commands and echo

- Change greeting message
- Read user command and echo

Santosh3007 committed:
feat: store and display input texts

- Store texts entered by user and display them when requested

Santosh3007 committed:
refactor: add task class

Santosh3007 committed:
feat: add/change status for tasks

- Add status to Task class
- Allow user to mark tasks as done / not done

Santosh3007 committed:
feat: create todos, events and deadlines

- allow users to create 3 different types of tasks: todos, events and deadlines
- create these new classes inheriting from task and make task class abstract

Santosh3007 committed:
chore: configure text-ui-testing

- write tests for text-ui-testing
- change ACTUAL.txt file name in .gitignore

Santosh3007 committed:
feat: create DukeException class

- create DukeException class to handle errors related to Duke

Santosh3007 committed:
refactor: add factory methods to create tasks

- add factory methods in task, todo, deadline and event classes to create tasks

Santosh3007 committed:
feat: add error handling

- add error handling for wrong user inputs

Santosh3007 committed:
chore: add tests for error handling

Santosh3007 committed:
feat: allow tasks to be deleted

- add delete command to allow tasks to be deleted
- add changeTaskStatus function to reduce duplicate code

Santosh3007 committed:
chore: add tests for delete tasks

- add tests for deleting tasks
- format styling of output text

Santosh3007 committed:
refactor: use enums

- use enums to represent valid user commands

Santosh3007 committed:
chore: add javadocs

Santosh3007 committed:
refactor: refactor DukeException class

- use message declared in Exception and use getMessage() to get the error message
- fix test cases after changing error messages

Santosh3007 committed:
style: follow style guide

- change style to follow Java coding standards

Santosh3007 committed:
feat: allow tasks to be created with isDone

- allow tasks to be created with isDone specified
- add getFileString() function to form string which will be stored locally

Santosh3007 committed:
feat: save tasks to hard disk

- save tasks to hard disk when closing program
- load tasks from hard disk if file exists when starting program

Santosh3007 committed:
feat: store deadline and events as dates

- store deadlines and events as dates
- accept different formats of dates as input
- format the date to make it more readable before printing

Santosh3007 committed:
test: change date formats for test cases

- change date formats for test cases
- add test cases to check date formats

Santosh3007 committed:
Merge branch 'branch-Level-7'

Santosh3007 committed:
Merge branch 'branch-Level-8'

Santosh3007 committed:
fix: allow dates to be saved

- fix buggy parseDate method
- change format when storing dates to file
- parse dates back when reading form file

Santosh3007 committed:
fix: fix datafile creation bug

- fix bug when creating new data directory
- fix styling for event toString()
- change test script to delete data.txt before run

Santosh3007 committed:
feat: create new UI class

- create Ui class and move methods related to text ui from duke

Santosh3007 committed:
test: add testing for storage file

- add tests for file saving and storage

Santosh3007 committed:
feat: create storage class

- create storage class to handle methods related to saving and reading from local storage
- move logic into run() method and call it from a new Duke instance in main method

Santosh3007 committed:
feat: create TaskList class

- create taskList class to handle logic related to managing tasks

Santosh3007 committed:
feature: add command classes and parser class

- extract command classes to handle the logic for each user command
- add parser class to deal with making sense of the user comman

Santosh3007 committed:
feature: divide classes into packages

- divide classes into separate packages: command, task, util

Santosh3007 committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

Santosh3007 committed:
feature: add checkstyle gradle task

- add checkstyle gradle task

Santosh3007 committed:
test: add junit test classes for testing

- add test classes for Task and Parser class using junit

Santosh3007 committed:
feature: package app as jar file

- package app as jar file using shadowjar

Santosh3007 committed:
feature: add javadocs

- add javadocs for classes and methods

Santosh3007 committed:
style: fix coding standard violations

- fix coding standard violations

Santosh3007 committed:
fix: validate input when changing task status

- check whether task index is provided when changing task status

Santosh3007 committed:
feature: allow users to find tasks

- allow users to find task using the task description
- fix parserTest tests after changing ChangeStatusCommand constructor

Santosh3007 committed:
Merge branch 'branch-A-JavaDoc'

Santosh3007 committed:
Merge branch 'branch-A-CodingStandard'

Santosh3007 committed:
Merge branch 'branch-Level-9'

linuschancs commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,37 @@
+ * Represents a command to add a task to the task list.
+ */
+public class AddCommand extends Command {
+    String input;
```
Missing access modifier here!
```suggestion
    private String input;
```

linuschancs commented on src/main/java/duke/command/ChangeStatusCommand.java:
```diff
@@ -0,0 +1,55 @@
+ * Represents a command to change the status of task in the task list.
+ */
+public class ChangeStatusCommand extends Command {
+    int taskIndex;
+    boolean isDone;
```
Good naming conventions were used however I noticed there are missing access modifiers here as well as other places in your other classes. Perhaps a thought to consider would be if no access modifier is stated, what would it default to, and whether that is desirable?

linuschancs commented on src/main/java/duke/command/ChangeStatusCommand.java:
```diff
@@ -0,0 +1,55 @@
+        ChangeStatusCommand other = (ChangeStatusCommand) o;
+        return other.taskIndex == this.taskIndex && other.isDone == this.isDone;
+    }
+}
```
Very minor thing of missing a newline at the end of the file.

linuschancs commented on src/main/java/duke/command/Command.java:
```diff
@@ -0,0 +1,40 @@
+/**
+ * Represents a user command and its functionalities.
+ */
+public abstract class Command {
```
I really like your use of abstract classes to abstract the different types of commands!

linuschancs commented on src/main/java/duke/util/TaskList.java:
```diff
@@ -0,0 +1,119 @@
+ * Represents a list of tasks.
+ */
+public class TaskList {
+    ArrayList<Task> tasks;
```
Just a minor point of consideration, personally it may feel a little confusing as 'tasks' and 'task' names are used to represent the ArrayList and individual tasks respectively, perhaps changing the naming for 'tasks' into 'taskList' may improve the distinction of the two variables!

linuschancs reviewed (commented):
Overall I really love how you have structured your code and kudos to your JavaDocs, you have really written it very concisely and meticulously! Good coding convention was also practiced for most of your code! The only main minor consideration would be to be careful not to miss out your access modifiers when declaring class variables! Really good job! 👍

Santosh3007 committed:
feature: add GUI for duke

- add JavaFX
- support GUI interface

Santosh3007 committed:
Merge branch 'branch-Level-10'

Santosh3007 committed:
feature: add assertions

- add assertions in functions for validations

Santosh3007 committed:
style: remove unused code blocks

- remove unused code blocks after removing text based UI

Santosh3007 committed:
style: fix code quality issues

- fix code quality issues
- change javadoc comments to follow convention

Santosh3007 committed:
Merge pull request #5 from Santosh3007/branch-A-Assertions

Add Assertions

Santosh3007 committed:
Merge pull request #4 from Santosh3007/branch-A-CodeQuality

Code Quality Changes

Santosh3007 committed:
Add Help Command

Users might not know the list of commands when starting to
use the application

- Add another help command to list all available commands to users and
their respective syntaxes
- Prompt the user to type help if needed when entering invalid command

Santosh3007 committed:
Merge branch 'branch-C-Help'

Santosh3007 committed:
style: make gui better

- adjust main window sizing
- change background color and text color

Santosh3007 committed:
style: add personality for cerebro

- format greeting message
- add cerebro image

Santosh3007 committed:
fix: add new line for new task

- text displayed when adding new task does not have new line
- add new line to the return text

Santosh3007 committed:
feature: add ui image to docs

- add ui image to docs folder

Santosh3007 committed:
feature: update readme

- update readme in docs folder to show user guide

Santosh3007 committed:
feature: create jar file for v0.2

- create jar file for v0.2
- adjust window height to support mac
- font styles do not work on mac

Santosh3007 committed:
style: add javadocs

- add javadocs for methods without any javadocs
- update javadocs for methods returning string after changing to GUI

Santosh3007 committed:
style: fix Checkstyle errors

- fix checkstyle errors

Santosh3007 committed:
fix: add access modifiers

- add access modifiers missing for variables that are missing it

Santosh3007 committed:
Update build.gradle file

- Support cross platform builds
